packages/devtools/src/utils.ts

```diff
@@ -23,3 +23,24 @@ export function getRulesBySlots(node: DebugResult, result: SlotInfo[] = []): Slo
     return [...acc, ...getRulesBySlots(child)];
   }, result);
 }
+
+export function filterSlots(slots: SlotInfo[], searchTerm: string) {
```

Comment:
@YuanboXue-Amber you don't like `.reduce()`, right? 🐱 

```ts
  return slots.reduce<SlotInfo[]>((acc, { slot, rules }) => {
    const filteredRules = rules.filter(rule => rule.cssRule.includes(searchTerm));

    if (filteredRules.length) {
      return [...acc, { slot, rules: filteredRules }];
    }

    return acc;
  }, []);
```

Reply:
Haha yeah I didn't think of this is such a nice use case for reduce. Changes applied :)